### PR TITLE
Journal dlopen cleanup and qrcode printing test

### PR DESCRIPTION
**Pull request by keszybz**
Split out from #17453.

### CODE REVIEW

**Comment by keszybz** [comment]
```
20:09:11 === Failed test log ===
20:09:11 --- test-qrcode-util begin ---
20:09:11 Found container virtualization none.
20:09:11 QRCODE support is not installed: libqrencode.so.4: cannot open shared object file: No such file or directory
20:09:11 Failed to print QR code: Operation not supported
20:09:11 --- test-qrcode-util end ---
```
Hmmm.

**Comment by keszybz** [comment]
I made the test return "skipped" when we're in a non-utf8 locale... Let's see.

**Comment by keszybz** [comment]
```
08:03:54 --- test-qrcode-util begin ---
08:03:54 Found container virtualization none.
08:03:54 QRCODE support is not installed: libqrencode.so.4: cannot open shared object file: No such file or directory
08:03:54 Failed to print QR code: Operation not supported
08:03:54 --- test-qrcode-util end ---
```
Arghhhh!

**Comment by keszybz** [comment]
bionic-i386 timed out.
lgtm failed to build "base": The analysis for the base commit (d0c4275) was not successful.

File: src/journal/journalctl.c [hunk 1]
```diff
@@ -1779,6 +1779,53 @@ static int add_syslog_identifier(sd_journal *j) {
         return 0;
 }
 
+static int format_journal_url(
```
Comment (vcaputo) [inline src/journal/journalctl.c:1782]: FYI:
```
[612/1478] Compiling C object journalctl.p/src_journal_journalctl.c.o
../src/journal/journalctl.c:1782:12: warning: ‘format_journal_url’ defined but not used [-Wunused-function]
 1782 | static int format_journal_url(
      |            ^~~~~~~~~~~~~~~~~~
```

```
         enabled features: IMA, SELinux, SMACK, lz4, idn, initrd, binfmt, vconsole, quotacheck, tmpfiles, environment.d, sysusers, firstboot, randomseed, backlight, rfkill, xdg-autostart, logind, machined, portabled, userdb, hostnamed, timedated, timesyncd, localed, networkd, resolve, coredump, pstore, oomd, polkit, efi, pcre2, blkid, nss-myhostname, nss-mymachines, nss-resolve, nss-systemd, hwdb, tpm, SysV compat, utmp, ldconfig, hibernate, adm group, wheel group, gshadow, link-udev-shared, link-systemctl-shared, link-networkd-shared, link-timesyncd-shared, kernel-install, systemd-analyze

         disabled features: libcryptsetup, PAM, pwquality, libfdisk, p11kit, libfido2, AUDIT, AppArmor, SECCOMP, zlib, xz, zstd, bzip2, ACL, gcrypt, qrencode, microhttpd, gnutls, openssl, libcurl, compat-mutable-uid-boundaries, libidn2, libidn, libiptc, elfutils, repart, homed, importd, DNS-over-TLS(gnutls), DNS-over-TLS(openssl), legacy pkla, gnu-efi, kmod, xkbcommon, dbus, glib, man pages, html pages, man page indices, debug hashmap, debug mmap cache, debug siphash, valgrind, trace logging, install tests
```

Comment (yuwata) [reply src/journal/journalctl.c:1782]: Fix is waiting in #17486.

**Pull request merged**